paymaster-analysis/alchemy/alchemy-account.ts

```diff
@@ -0,0 +1,290 @@
+    .catch((err) => console.error(err));
+} else if (chain == "goerli") {
+  await fetch("https://eth-goerli.g.alchemy.com/v2/" + apiKey, gasOptions)
+    .then((response) => response.json())
+    .then((response) => (responseValues = response))
+    .catch((err) => console.error(err));
+} else {
+  throw new Error(
+    "Current code only support Sepolia and Goerli. Please make required changes if you want to use custom network.",
+  );
+}
+console.log("\nReceived Paymaster Data from Alchemy.");
+
+sponsoredUserOperation.preVerificationGas =
+  responseValues.result.preVerificationGas;
+sponsoredUserOperation.preVerificationGas =
+  responseValues.result.preVerificationGas;
+sponsoredUserOperation.callGasLimit = responseValues.result.callGasLimit;
+sponsoredUserOperation.verificationGasLimit =
+  responseValues.result.verificationGasLimit;
+sponsoredUserOperation.paymasterAndData =
+  responseValues.result.paymasterAndData;
+sponsoredUserOperation.maxFeePerGas = responseValues.result.maxFeePerGas;
+sponsoredUserOperation.maxPriorityFeePerGas =
+  responseValues.result.maxPriorityFeePerGas;
+
+sponsoredUserOperation.signature = await signUserOperation(
+  sponsoredUserOperation,
+  signer,
+);
+console.log("\nSigned Real Data including Paymaster Data Created by Alchemy.");
+
+// console.log(sponsoredUserOperation);
+
+const options = {
+  method: "POST",
+  headers: { accept: "application/json", "content-type": "application/json" },
+  body: JSON.stringify({
+    id: 1,
+    jsonrpc: "2.0",
+    method: "eth_sendUserOperation",
+    params: [
+      {
+        sender: sponsoredUserOperation.sender,
+        nonce: "0x" + sponsoredUserOperation.nonce.toString(16),
+        initCode: sponsoredUserOperation.initCode,
+        callData: sponsoredUserOperation.callData,
+        callGasLimit: sponsoredUserOperation.callGasLimit,
+        verificationGasLimit: sponsoredUserOperation.verificationGasLimit,
+        preVerificationGas: sponsoredUserOperation.preVerificationGas,
+        maxFeePerGas: sponsoredUserOperation.maxFeePerGas,
+        maxPriorityFeePerGas: sponsoredUserOperation.maxPriorityFeePerGas,
+        signature: sponsoredUserOperation.signature,
+        paymasterAndData: sponsoredUserOperation.paymasterAndData,
+      },
+      ENTRY_POINT_ADDRESS,
+    ],
+  }),
+};
+
+if (chain == "sepolia") {
+  await fetch("https://eth-sepolia.g.alchemy.com/v2/" + apiKey, options)
+    .then((response) => response.json())
+    .then((response) => (responseValues = response))
+    .catch((err) => console.error(err));
+} else if (chain == "goerli") {
+  await fetch("https://eth-goerli.g.alchemy.com/v2/" + apiKey, options)
+    .then((response) => response.json())
+    .then((response) => (responseValues = response))
+    .catch((err) => console.error(err));
+} else {
+  throw new Error(
+    "Current code only support Sepolia and Goerli. Please make required changes if you want to use custom network.",
+  );
```

Comment:
nit; Can't we check for supported chains once early in the file and then:

```typescript
  await fetch(`https://eth-${network}.g.alchemy.com/v2/${apiKey}`, options)
```

This would also apply to the Etherscan links for example.

Reply:
Yes, we can do that 👍🏾

Reply:
Done 👍🏾